Calculate/Program.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+
+class Program
+{
+    static void Main()
+    {
+        var calculator = new Calculator();
+        var program = new Program();
+        program.SetUpInput(Console.WriteLine, Console.ReadLine);
+        program.Run(calculator);
+    }
+
+    public Action<string> WriteLine { get; set; }
+    public Func<string> ReadLine { get; set; }
+    public void SetUpInput(Action<string> writeLine, Func<string> readLine)
+    {
+        WriteLine = writeLine;
+        ReadLine = readLine;
+    }
```

Comment:
```suggestion
    public Program(Action<string> writeLine, Func<string> readLine)
    {
        WriteLine = writeLine;
        ReadLine = readLine;
    }

    public Program()
    {
        WriteLine = Console.WriteLine;
        ReadLine = Console.ReadLine;
    }
```
You should really be doing something line this